bench_test.go

```diff
@@ -112,6 +122,60 @@ func BenchmarkSet(b *testing.B) {
 	})
 }
 
+func BenchmarkSetIfPresent(b *testing.B) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	tab := []struct {
+		name string
+		imp  Geche[string, string]
+	}{
+		{
+			"MapCache",
+			NewMapCache[string, string](),
+		},
+		{
+			"StringCache",
+			newStringCache(),
+		},
+		{
+			"UnsafeCache",
+			newUnsafeCache(),
+		},
+		{
+			"MapTTLCache",
+			NewMapTTLCache[string, string](ctx, time.Minute, time.Minute),
+		},
+		{
+			"RingBuffer",
+			NewRingBuffer[string, string](1000000),
+		},
+		{
+			"KVMapCache",
+			NewKV[string](NewMapCache[string, string]()),
+		},
+	}
+
+	data := genTestData(10_000_000)
+
+	for _, c := range tab {
+		b.Run(c.name, func(b *testing.B) {
+			benchmarkSetIfPresent(c.imp, data, b)
+		})
+	}
+
+	b.Run("AnyCache", func(b *testing.B) {
```

Comment:
This one is not needed. `anyCache` is not exported and just exists to evaluate performance impact of empty interface conversion compared to generic typed version.